### PR TITLE
Turnstile Invisible Support for Placeholder

### LINKED ISSUE #8: Disable Predetermined Component Sizing for Invisible Turnstile
Continuing from #7,  I think we can add a new option for `size: invisible`. The option is just for this library to disable the predetermined height/width, and will not be forwarded to Turnstile.

[Invisible Turnstile](https://developers.cloudflare.com/turnstile/reference/widget-types/#invisible) will not show any widget at all. So the predetermined size is actually reserving a space that will never be used. This newly introduced option will help preventing that.

### PR DESCRIPTION
**Pull request by chez14**
In this PR:
- Introduce `size: invisible` to prevent the component from creating a placeholder for the widget.
- Updated the documentation for the size `RenderOptions`.

More context for the changes:
- Since `size: invisible` is consumed just for this component only, the changes will just not set any width/height at all when the size is set to this option. The parameter will not be forwarded to Turnstile's code too.

This PR fixes #8.

Thank you!

### CODE REVIEW

**Comment by chez14** [comment]
The only problem current PR is... I can't update the demo because Turnstile doesn't have any invisible testing key. I also cannot add unit test for this for the same reason.

I wonder if there's better approach for this.

**Comment by marsidev** [comment]
Hey, thanks for your contribution, looks good. Did you tested it locally? I tried adding a "invisible" option for the demo but the widget still renders. 

image

Seems when the invisible option is selected, the size parameter sent to cf is `undefined`, which causes to render a "normal" widget by default.

What do you think about adding a `display: none` in the widget container if `size === 'invisible'`? This would still render a normal-sized widget but visually hidden.

**Comment by chez14** [comment]
Hi @marsidev,

After checking further with the widgets, apparently `invisible` means `non-interactive` with zero width and height. The one that sets them to zero width and height are the JS, the content are still simillar with the non-interactive type. :laughing:

I made a little test with v0.0.5 to demonstrate my findings: https://react-turnstile-test.playground.christianto.net/ (the code is on https://gitlab.com/chez14/react-turnstile-test). I'll explain more after answering your questions.

---

> I tried adding a "invisible" option for the demo but the widget still renders.

This is the problem [I mentioned](https://github.com/marsidev/react-turnstile/pull/9#issuecomment-1378702710) earlier, there is no testing key for invisible widget, so I can't just test them with your demo page, without adding my own personal keys. :disappointed:

> Seems when the invisible option is selected, the size parameter sent to cf is undefined, which causes to render a "normal" widget by default.

This is expected. In their [docs](https://developers.cloudflare.com/turnstile/get-started/client-side-rendering/#configurations), Turnstile's `size` option only accept `normal` and `compact`. This is why I mentioned that if `size` is `invisible`, the `size` option will not be forwarded to the Js... so its kind of "Use the default config whatever Cloudflare sets, just for invisible widget". 

Also, this is why I add the option in the `ComponentRenderOptions`, not the `RenderOptions`, because Turnstile doesn't actually accept `size: invisible`. This option is used by the lib only.

> What do you think about adding a display: none in the widget container if size === 'invisible'? This would still render a normal-sized widget but visually hidden.

I'm afraid that this will cause some issues with other browser. For example `img` with `display: none`, the image file will not be loaded by browser. After searching more information about this, Firefox actually have potential to have this problem: https://stackoverflow.com/a/21262365/4721245. (...though its still in debate).

---

Because of those, I try to check with [mini playground](https://react-turnstile-test.playground.christianto.net/) i mentioned earlier, and here's my findings:
- You can actually change the widget type in CF panel, I originally thought it was locked.
- Turnstile with Invisible widget type is actually just non-interactive widget. 
  The JS that fetched the widget info just set them to zero width and height. (To try, check the "Invisible" and "Invisible + force resize").
  Both key are Turnstile with Invisible key widget.


So I think it's better to set height and width to zero? And also don't set `display: none;` just to make sure browser behaves as we expected.

And for testing, I think we can use the non-interactive keys for the invisible size.

What do you think, Luis? If it's all okay, I'll apply the changes to current PR, or just let me know if you have any questions or feedback. Thank you for reviewing my PR!

**Comment by marsidev** [comment]
Hi @chez14, 

I have no created turnstile production keys in a while, so I forgot that the invisibility is defined by the widget type setting in the CF dashboard. So I now understand what you mean with "Turnstile doesn't have any invisible testing key" 😅

I'm agree to explicity set the width and height to 0 when we choose the `size: invisible` option.

For the demo page, could be great to have an Invisible option as I shown previously. And if this option is selected, remove the "Force interactive challenge" in the sitekey type. What do you think?

In that way, the size options would have the following behavior:

- `size: normal` - `siteKey: "Always pass" | "Always fail" | "Force interactive challenge"`
- `size: compact` - `siteKey: "Always pass" | "Always fail" | "Force interactive challenge"`
- `size: invisible` - `siteKey: "Always pass" | "Always fail"` (interactive challenge has no sense here)

If you are agree and you can handle that, go ahead :)

Thanks in advance.

**Comment by chez14** [comment]
Hi @marsidev,

> For the demo page, could be great to have an Invisible option as I shown previously. And if this option is selected, remove the "Force interactive challenge" in the sitekey type. What do you think?

Sounds awesome, I'll try to implement this then, just let me know if i need to update the patch somewhere. Thank you!

File: packages/example/src/components/ConfigForm.tsx [hunk 1]
```diff
@@ -7,41 +7,76 @@ interface FormProps {
...
 					helperUrl='https://developers.cloudflare.com/turnstile/frequently-asked-questions/#are-there-sitekeys-and-secret-keys-that-can-be-used-for-testing'
 					name='siteKey'
-					options={[
-						{ label: 'Always pass', value: 'pass' },
-						{ label: 'Always fail', value: 'fail' },
-						{ label: 'Force interactive challenge', value: 'interactive' }
-					]}
+					options={SiteKeyOptions.map(option => ({
+						...option,
+						// Option will be disabled on Invisibile widget type, and requesting
```
Comment (marsidev) [inline packages/example/src/components/ConfigForm.tsx:72]: Here is a typo _**Invisibile**_. Can you fix it?

**Review by marsidev — CHANGES_REQUESTED** [review]

**Comment by chez14** [comment]
Hi @marsidev,

Sorry for the delayed response, I've reworded the comments because when I read it can be reworded better.

Thank you!

**Review by marsidev — APPROVED** [review]

**Comment by marsidev** [comment]
Thank you! Great work. 🎉  @chez14

**Pull request merged**